File: Ajedrez/app/src/main/java/com/example/ajedrez/View/Lessons/LessonsListAdapter.java
package com.example.ajedrez.View.Lessons;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.ajedrez.Model.Lesson;
import com.example.ajedrez.R;
import com.example.ajedrez.View.Lessons.LessonsListFragment.LessonsListener;

import java.util.List;

public class LessonsListAdapter extends RecyclerView.Adapter<LessonsListAdapter.ViewHolder> {

    private List<Lesson> mValues;
    private final LessonsListener mListener;

    LessonsListAdapter(List<Lesson> items, LessonsListener listener) {
        mValues = items;
        mListener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.fragment_lessons_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, int position) {
        Lesson lesson = mValues.get(position);
        holder.mItem = lesson;
        holder.mDate.setText(lesson.getDate());
        holder.mAttendance.setText(String.valueOf(lesson.getNumberOfAssists()));
        holder.mView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (null != mListener) {

                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return mValues.size();
    }

    void setLessonsList(List<Lesson> lessonsList) {
        this.mValues = lessonsList;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        final View mView;
        final TextView mDate;
        final TextView mAttendance;
        Lesson mItem;

        ViewHolder(View view) {
            super(view);
            mView = view;
            mDate = view.findViewById(R.id.item_number);
            mAttendance = view.findViewById(R.id.content);
        }

        @NonNull
        @Override
        public String toString() {
            return super.toString() + " '" + mAttendance.getText() + "'";
        }
    }
}

File: Ajedrez/app/src/main/java/com/example/ajedrez/View/Lessons/LessonsListAdapter.kt
package com.example.ajedrez.View.Lessons

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.example.ajedrez.Model.Lesson
import com.example.ajedrez.R
import com.example.ajedrez.View.Lessons.LessonsListFragment.LessonsListener

class LessonsListAdapter internal constructor(private var mValues: List<Lesson>,
                                              private val mListener: LessonsListener?) : RecyclerView.Adapter<LessonsListAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.fragment_lessons_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val lesson = mValues[position]
        holder.mItem = lesson
        holder.mDate.text = lesson.date
        holder.mAttendance.text = lesson.numberOfAssists.toString()
        holder.mView.setOnClickListener {
            mListener?.showLessonInfoScreen(lesson)
        }
    }

    override fun getItemCount(): Int {
        return mValues.size
    }

    fun setLessonsList(lessonsList: List<Lesson>) {
        mValues = lessonsList
    }

    inner class ViewHolder internal constructor(val mView: View) : RecyclerView.ViewHolder(mView) {

        val mDate: TextView = mView.findViewById(R.id.item_number)
        val mAttendance: TextView = mView.findViewById(R.id.content)
        var mItem: Lesson? = null

        override fun toString(): String {
            return super.toString() + " '" + mAttendance.text + "'"
        }

    }

}